src/main/webapp/gast/stat.jsp

```diff
@@ -0,0 +1,155 @@
+    if (request.getParameter("currentPage") != null) {
+        currentPage = Integer.parseInt(request.getParameter("currentPage"));
+    }
+
+    int recordsPerPage = Constants.RECORDS_PER_PAGE;
+
+    if (request.getParameter("recordsPerPage") != null) {
+        recordsPerPage = Integer.parseInt(request.getParameter("recordsPerPage"));
+    }
+
+    int rows = QuelleDB.countStat(filterTitle).intValue();
+
+    //row count is 1 if users performs id search
+    if(jumpToID != null && jumpToID.length() > 0){
+        rows = 1;
+    }
+    
+    int nOfPages = rows / recordsPerPage;
+
+    if (nOfPages % recordsPerPage > 0) {
+        nOfPages++;
+    }
+    
+    List<Quelle> lst = QuelleDB.getList(currentPage, recordsPerPage,filterTitle,sort, jumpToID);
+%>
+<%!
+        
+    /*
+    Functions that helps to print pagination
+     */
+    public String html_prev_button(String filterTitle, String recordsPerPage, Integer currentPage, String sort, String jumpToID) {
+        return "<li class=\"page-item\"><a class=\"page-link\" href=\"?jumpToID="+jumpToID+"&sort="+sort+"&page=anzahl_belege&filterTitle=" + filterTitle + "&recordsPerPage=" + recordsPerPage + "&currentPage=" + (currentPage - 1) + "\">Previous</a></li>";
+    }
+
+    public String html_next_button(String filterTitle, String recordsPerPage, Integer currentPage, String sort, String jumpToID) {
+        return "<li class=\"page-item\"><a class=\"page-link\"href=\"?jumpToID="+jumpToID+"&sort="+sort+"&page=anzahl_belege&filterTitle=" + filterTitle + "&recordsPerPage=" + recordsPerPage + "&currentPage=" + (currentPage + 1) + "\">Next</a></li>";
+    }
+
+    public String html_page_item_current(int page) {
+        return "<li class=\"page-item active\"><a class=\"page-link\">" + page + "</a></li>";
+    }
+
+    public String html_page_item(int page, String filterTitle, int recordsPerPage, String sort, String jumpToID) {
+        return "<li class=\"page-item\"><a class=\"page-link\"href=\"?jumpToID="+jumpToID+"&sort="+sort+"&page=anzahl_belege&filterTitle=" + filterTitle + "&recordsPerPage=" + recordsPerPage + "&currentPage=" + page + "\">" + page + "</a></li>";
+    }
+
+    public String html_sort_title_up(int page, String filterTitle, int recordsPerPage, String jumpToID) {
+        return "<a class=\"sort-link\"href=\"?jumpToID="+jumpToID+"&sort=titleUp&page=anzahl_belege&filterTitle=" + filterTitle + "&recordsPerPage=" + recordsPerPage + "&currentPage=" + page + "\">sort up</a>";
+    }
+
+    public String html_sort_title_down(int page, String filterTitle, int recordsPerPage, String jumpToID) {
+        return "<a class=\"sort-link\"href=\"?jumpToID="+jumpToID+"&sort=titleDown&page=anzahl_belege&filterTitle=" + filterTitle + "&recordsPerPage=" + recordsPerPage + "&currentPage=" + page + "\">sort down</a>";
+    }
+
+    public void print_pagination(JspWriter out, int currentPage, int recordsPerPage, String filterTitle, int nOfPages, String sort, String jumpToID) throws Exception {
```

Comment:
Instead of the custom pagination elements => can we re-use the pagination that is also used in other templates?

Since this is a list, i understand that including the regular title.inc.jsp might not make sense, but maybe you can include forms/link.jsp in a similar way?

Reply:
I think it is hard to reuse forms/link.jsp. 
link.jsp uses sql queries to determine the ID of the next and previous dataset,
while html_prev_button and html_next_button is based on pagination which calculate the next and previous page, based on the current page.